### DISCUSSION TITLE
Store Templates in NOT the browser cache? :(

### DISCUSSION THREAD

**Discussion by poisenbery** [comfyanonymous/ComfyUI#2159] [Ideas]
So...this is obviously my fault....but all of my node templates are gone.

Why: https://github.com/comfyanonymous/ComfyUI/issues/1961

I Incorrectly assumed that the templates were saved to a file somewhere in the comfyUI directory, or some type of .json

I had a problem with my browser and had to clear cache, so all of my templates are gone :( :( :( 

Can this...be changed?

I know, I should have exported them and saved them, but I literally thought that they were already saved and safe (which is my fault).

**Comment by Woisek** [comment]
The templates can be imported/exported ... why didn't you export them at least once ... ?

**Reply by dnrghostfox** [reply]
我也是同样原因，因为清楚了缓存后就找不到保存的预设了，可是我的user-default文件夹下面的comfy.templates.json文件中是有保存的预设内容的，但浏览器中却显示一片空白，用加载功能导入那个comfy.templates.json文件也没有反应

**Reply by dnrghostfox** [reply]
I have the same reason, because I can't find the saved presets after clearing the cache, but I have the saved presets in the comfy.templates.json file under the user-default folder, but the browser shows a blank. Importing the comfy.templates.json file with the load function also didn't respond

**Reply by ltdrdata** [reply]
In the new UI, workflows are managed internally in the system's own storage. It provides functionality to paste selected workflows into the current workflow like templates, which can replace the function of templates.

**Comment by Burnrate** [comment]
This happened to me a few times and I thought it was a bug or I was doing something wrong with my templates.  Very frustrating.  Such an odd choice to store them in the browser cache.  I understand you can export them but idk it seems so inconvenient.

**Comment by comfyanonymous** [comment]
https://github.com/comfyanonymous/ComfyUI/pull/2160

Since this you can use --server-storage which will be enabled by default in the very near future.

**Reply by worthy7** [reply]
Yes please, I want to use my templates on any workflow by default!
